test/Core.Test/AdminConsole/OrganizationAuth/UpdateOrganizationAuthRequestCommandTests.cs

```diff
@@ -76,25 +78,143 @@ await sutProvider.GetDependency<IMailService>().DidNotReceive()
+            Arg.Any<string>(),
+            Arg.Any<DateTime>(),
+            Arg.Any<string>(),
+            Arg.Any<string>()
+        );
+        await sutProvider.GetDependency<IEventService>().ReceivedWithAnyArgs().LogOrganizationUserEventsAsync(
+            Arg.Any<IEnumerable<(OrganizationUser, EventType, DateTime?)>>()
+        );
+    }
+
+    [Theory]
+    [BitAutoData]
+    public async Task UpdateAsync_BatchUpdate_AuthRequestIsDenied_DoesNotLeakRejection(
+        SutProvider<UpdateOrganizationAuthRequestCommand> sutProvider,
+        List<OrganizationAuthRequestUpdate> updates,
+        OrganizationAdminAuthRequest unprocessedAuthRequest,
+        AuthRequestUpdateProcessorConfiguration configuration,
+        User user
+    )
+    {
+        // For this command to work we need the following from external
+        // classes:
+        // 1. A configured expiration timespan for organization auth requests
+        // 2. Some unresponded to auth requests that match the ids provided
+        // 3. A valid user to send emails to
+
+        var unprocessedAuthRequests = new List<OrganizationAdminAuthRequest>();
+        unprocessedAuthRequest = UnrespondAndEnsureValid(unprocessedAuthRequest, configuration.OrganizationId);
+        foreach (var update in updates)
+        {
+            update.Approved = false;
+            unprocessedAuthRequest.Id = update.Id;
+            unprocessedAuthRequests.Add(unprocessedAuthRequest);
+        };
+
+        sutProvider.GetDependency<IGlobalSettings>().PasswordlessAuth.AdminRequestExpiration.Returns(TimeSpan.FromDays(7));
+
+        sutProvider.GetDependency<IAuthRequestRepository>().GetManyAdminApprovalRequestsByManyIdsAsync(
+            configuration.OrganizationId,
+            updates.Select(ar => ar.Id)
+        ).ReturnsForAnyArgs(unprocessedAuthRequests);
+
+        sutProvider.GetDependency<IUserRepository>().GetByIdAsync(Arg.Any<Guid>()).Returns(user);
+
+        // Call the SUT
+        await sutProvider.Sut.UpdateAsync(configuration.OrganizationId, updates);
+
+        // Assert that because we passed in good data we call a save
+        // operation and raise all events
+        await sutProvider.GetDependency<IAuthRequestRepository>().ReceivedWithAnyArgs().UpdateManyAsync(Arg.Any<IEnumerable<OrganizationAdminAuthRequest>>());
+        await sutProvider.GetDependency<IPushNotificationService>().DidNotReceiveWithAnyArgs().PushAuthRequestResponseAsync(Arg.Any<AuthRequest>());
+        await sutProvider.GetDependency<IMailService>().DidNotReceiveWithAnyArgs().SendTrustedDeviceAdminApprovalEmailAsync(
+            Arg.Any<string>(),
+            Arg.Any<DateTime>(),
+            Arg.Any<string>(),
+            Arg.Any<string>()
+        );
+        await sutProvider.GetDependency<IEventService>().ReceivedWithAnyArgs().LogOrganizationUserEventsAsync(
+            Arg.Any<IEnumerable<(OrganizationUser, EventType, DateTime?)>>()
+        );
```

Comment:
Please use `Arg.Is` to assert some basic values here - I think that asserting calls with `Arg.Any` is not very precise. And while the processor tests test that the callbacks are called, you want to make sure that the callback has passed on the arguments to the dependency correctly.

Reply:
I didn't really know how to do this, but appreciate the push to do it because it was fun. The test is more precise as of https://github.com/bitwarden/server/pull/4064/commits/51a837a685d4def608cb6fff4197f592887786c1.